io/disk/disktest.py

```diff
@@ -0,0 +1,115 @@
+import os
+import sys
+import subprocess
+import logging
+
+
+from avocado import Test
+from avocado import main
+from avocado.utils import memory
+from avocado.utils import disk
+from avocado.utils import process
+from avocado.utils import build
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Disktest(Test):
+
+    """
+    Avocado module for disktest.
+    Pattern test of the disk, using unique signatures for each block and each
+    iteration of the test. Designed to check for data corruption issues in the
+    disk and disk controller.
+    It writes 50MB/s of 500KB size ops.
+    """
+    version = 2
+
+    def setUp(self):
+        """
+        Verifies if we have gcc to compile disktest.
+        """
+        softm = SoftwareManager()
+        if not softm.check_installed("gcc") and not softm.install("gcc"):
+            self.error('Gcc is needed for the test to be run')
+        self.disks = self.params.get('disks', default=None)
+        self.gigabytes = self.params.get('gigabytes', default=None)
+        self.chunk_mb = self.params.get('chunk_mb', default=None)
+        self.memory_mb = memory.memtotal() / 1024 / 8
+        self.data_dir = os.path.join(os.path.abspath(self.datadir))
+        self.disk_srcdir = os.path.join(os.path.abspath(self.datadir), 'src')
+        self.build(self.disk_srcdir)
+
+    def build(self, srcdir):
+        """
+        Compiles disktest.
+        """
+        os.chdir(srcdir)
+        build.make(srcdir, extra_args='clean')
+        build.make(srcdir)
+
+    def one_disk_chunk(self, disks, chunk):
+        """
+        Tests one part of the disk by spawning a disktest instance.
+        :param disks: Directory (usually a mountpoint).
+        :param chunk: Portion of the disk used.
+        """
+        logging.info("Testing %d MB files on %s in %d MB memory, chunk %s",
+                     self.chunk_mb, disks, self.memory_mb, chunk)
+        logfile = self.outputdir + '/log.txt'
+        cmd = ("%s/disktest -m %d -f %s/testfile.%d -i -S >>%s 2>&1" %
+               (self.disk_srcdir, self.chunk_mb, disks, chunk, logfile))
+        logging.debug("Running '%s'", cmd)
+        thread = subprocess.Popen(cmd, shell=True)
```

Comment:
avocado has a modified version for subprocesses in `avocado.utils.process`. It'd be better to use that one instead. The benefit is you can use wrappers/gdb/... dynamically from cmdline (used by developers).

```python
proc = process.get_sub_process_klass(cmd)(cmd, shell=True)
pid = proc.start()
...
proc.poll()
result = proc.wait()
```

This way one can use features like: http://avocado-framework.readthedocs.io/en/latest/DebuggingWithGDB.html and http://avocado-framework.readthedocs.io/en/latest/WrapProcess.html